### PR TITLE
[SDPSUP-1212] : Add word count form validation

### PR DESCRIPTION
**Pull request by FleetAdmiralButter**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPSUP-1212

### Changed

1.  Added custom form validation methods to check number of words in textArea and textField form elements if configured under the "Form Validation" tab in Drupal.

2.  Counter_maximum and counter_minimum values override maxlength and minlength if both are set.

3. Move to using VFG validation instead of browser validation for all free text fields (textArea, textField, tel, url, email)

### Screenshots

Screen Shot 2019-12-06 at 10 15 04 am
Screen Shot 2019-12-06 at 10 14 36 am
Screen Shot 2019-12-06 at 10 14 19 am

### CODE REVIEW

File: packages/components/Molecules/Form/index.vue [hunk 1]
```diff
@@ -71,6 +71,17 @@ export default {
   },
   mounted () {
     RplFormEventBus.$on('clearform', this.clearForm)
+    VueFormGenerator.validators.rplWordCount = function (value, field) {
```
Comment (tim-yao) [inline packages/components/Molecules/Form/index.vue]: Can you please add a `TODO` comment for this, we need to move custom validators into separated files in future. We will add unit test in that time.

File: packages/ripple-nuxt-tide/modules/webform/mapping-filters.js [hunk 2]
```diff
@@ -143,9 +163,28 @@ module.exports = {
...
+          // We add a hint to let the user know of the character limit.
+          if (field.max) {
+            if (field.counter_type !== 'word') {
+              let fieldMaxMessage = 'Character limit is ' + field.max
+
+              if (field.hint) {
+                field.hint += fieldMaxMessage
+              } else {
+                field.hint = fieldMaxMessage
+              }
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/modules/webform/mapping-filters.js]: You can use 
```
field.hint = field.hint ? `${field.hint} ${fieldMaxMessage}`: fieldMaxMessage
```
to reduce lines.

File: packages/ripple-nuxt-tide/modules/webform/mapping-filters.js [hunk 3]
```diff
@@ -143,9 +163,28 @@ module.exports = {
...
+
+          // If we're using string validation, VFG sets maxlength in the browser, and
+          // cuts off over limit text instead of displaying an error.
+          // We add a hint to let the user know of the character limit.
+          if (field.max) {
+            if (field.counter_type !== 'word') {
+              let fieldMaxMessage = 'Character limit is ' + field.max
+
+              if (field.hint) {
+                field.hint += fieldMaxMessage
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/modules/webform/mapping-filters.js]: Need a space when concat two strings.

**Comment by FleetAdmiralButter** [comment]
@tim-yao,

image

**Pull request merged**